model/openapi.yml

```
@@ -1251,3 +1251,44 @@ paths:
           $ref: '#/components/responses/internalServerError'
         '503':
           $ref: '#/components/responses/unavailable'
+  /api/fleet/file/{id}:
+    get:
+      operationId: getFile
+      summary: retrieve stored file for integration
+      description: "Stream out file contents to an agent or integration, provided there is a matching and authorized file stored in elasticsearch."
+      security:
+        - agentApiKey: []
+      parameters:
+        - name: id
+          in: path
+          description: The file_id as provided to the integration
+          required: true
+          schema:
+            type: string
+            example: ecb30383-6dd1-4b1d-bed0-2386b4e5df51
+      responses:
+        '200':
+          description: file contents
```

```suggestion
          description: File contents.
          content:
            "*/*":
              schema:
                type: string
                format: binary
```
(nitpick) I don't think this will change any of the generated code

Can you also add the `X-File-Sha2` header to the 200 response description?